internal/cli/atlas/alerts/settings/settings.go

```diff
@@ -101,49 +116,83 @@ func (opts *ConfigOpts) newNotification() *admin.AlertsNotificationRootForGroup
 		out.NotifierId = &opts.notifierID
 	}
 
+	// write set of functions that contain code from switch cases and return error if required fields are not provided
 	switch out.GetTypeName() {
 	case datadog:
+		if opts.apiKey == "" || opts.notificationRegion == "" {
+			return nil, errors.New("--apiKey and --notificationRegion are required when --notificationType is DATADOG")
+		}
```

Comment:
we normally don't tangle creation and validation together, can you please move the validations to a `validate` method and call it as part of the PreRun

Reply:
Done